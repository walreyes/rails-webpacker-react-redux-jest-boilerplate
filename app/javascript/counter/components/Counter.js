import React from 'react'
import PropTypes from 'prop-types'

export default class Counter extends React.Component {
  static propTypes = {
    counter     : PropTypes.number.isRequired,
    doubleAsync : PropTypes.func.isRequired,
    increment   : PropTypes.func.isRequired
  }

  render () {
    const counter = this.props.counter
    return (
      <div style={{ margin: '0 auto' }} >
        <h2>Counter: {counter}</h2>
        <button className='btn btn-default' onClick={this.props.increment}>
          Increment
        </button>
        {' '}
        <button className='btn btn-default' onClick={this.props.doubleAsync}>
          Double (Async)
        </button>
      </div>
    )
  }
}
