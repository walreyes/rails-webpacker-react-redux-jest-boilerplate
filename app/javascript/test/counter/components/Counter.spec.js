import React from 'react';
import Counter from '../../../counter/components/Counter';

const counterProps = {
  counter: 0,
  increment: jest.fn(),
  doubleAsync: jest.fn(),
}


it('should render Counter without crashing', () => {
  const wrapper = shallow(
      <Counter {...counterProps} />
  );

  expect(wrapper).toMatchSnapshot();
});

it('should render a counter 0', () => {
    const wrapper = mount(
        <Counter {...counterProps} />
    );

    expect(wrapper.prop('counter')).toEqual(0);
});
/*
it('should increment counter by 1', () => {
    const value = '2';
    const increment = jest.fn();
    const wrapper = shallow(
        <Counter counter={0} increment={increment} />
    );

    expect(wrapper).toMatchSnapshot();

        wrapper.find('select').simulate('change', {
        target: { value },
    });

    expect(onChange).toBeCalledWith(value);
});
*/




/*
it('should render a label', () => {
    const wrapper = shallow(
        <Label>Hello Jest!</Label>
    );
    expect(wrapper).toMatchSnapshot();
});

it('should render a small label', () => {
    const wrapper = shallow(
        <Label small>Hello Jest!</Label>
    );
    expect(wrapper).toMatchSnapshot();
});

it('should render a grayish label', () => {
    const wrapper = shallow(
        <Label light>Hello Jest!</Label>
    );
    expect(wrapper).toMatchSnapshot();
});

it('should render a document title', () => {
    const wrapper = shallow(
        <DocumentTitle title="Events" />
    );
    expect(wrapper.prop('title')).toEqual('Events');
});

it('should render a document title and a parent title', () => {
    const wrapper = shallow(
        <DocumentTitle title="Events" parent="Event Radar" />
    );
    expect(wrapper.prop('title')).toEqual('Events â€” Event Radar');
});

it('should render Markdown in preview mode', () => {
    const wrapper = shallow(
        <MarkdownEditor value="*Hello* Jest!" />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find('[name="toggle-preview"]').simulate('click');

    expect(wrapper).toMatchSnapshot();
});

it('should render Markdown in preview mode', () => {
    const wrapper = shallow(
        <MarkdownEditor value="*Hello* Jest!" />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find('[name="toggle-preview"]').simulate('click');

    expect(wrapper).toMatchSnapshot();
});
*/
