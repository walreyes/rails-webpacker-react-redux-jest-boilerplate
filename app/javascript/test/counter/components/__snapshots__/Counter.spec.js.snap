// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Counter without crashing 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
    }
  }
>
  <h2>
    Counter: 
    0
  </h2>
  <button
    className="btn btn-default"
    onClick={[Function]}
  >
    Increment
  </button>
   
  <button
    className="btn btn-default"
    onClick={[Function]}
  >
    Double (Async)
  </button>
</div>
`;
